timescaledb/tutorials/ingest-real-time-websocket-data.md

```diff
@@ -0,0 +1,347 @@
+    ``` 
+1. Install the Twelve Data Python [wrapper library](https://github.com/twelvedata/twelvedata-python) with websocket support:
+    ```bash
+    pip install twelvedata websocket-client
+    ```
+    This library makes it easy to make requests to the API and maintain a stable websocket connection.
+1. Install [Psycopg2][psycopg2] so that you can connect the
+    TimescaleDB from your Python script:
+
+    ```bash
+    pip install psycopg2-binary
+    ```
+
+</procedure>
+
+## Create the websocket connection
+When you connect to the Twelve Data API through a websocket, you create a
+persistent connection between your computer and the websocket server. This
+persistent connection can then be used to receive data as long as you or the
+server doesn't terminate the connection.
+
+<procedure>
+
+### Connecting to the websocket server
+
+1. Create a new Python file called `websocket_test.py` and connect to the
+    Twelve Data servers using the wrapper library:
+    
+    ```python
+    # websocket_test.py:
+    from twelvedata import TDClient
+    
+    def on_event(event):
+    	print(event) # prints out the data record (dictionary)
+    
+    td = TDClient(apikey="TWELVE_DATA_APIKEY")
+    ws = td.websocket(symbols=["BTC/USD", "ETH/USD"], on_event=on_event)
+    ws.connect()
+    ws.keep_alive()
+    ```
+    
+    Make sure to pass your API key as an argument for the `TDClient` object.
+    
+    As you can see in this code example, you need to pass two arguments to create
+    the websocket object:
+    
+    * **`on_event`**
+
+        This argument needs to be a function that gets invoked whenever there's a
+        new data record is received from the websocket.
+        ```python
+        def on_event(event):
+        	print(event) # prints out the data record (dictionary)
+        ```
+    
+        This is where you want to implement the ingestion logic so whenever
+        there's new data available you insert it into the database.
+    * **`symbol`**
+
+        This argument needs to be a list of stock ticker symbols (eg.: `MSFT`) or
+        crypto trading pairs (eg.: `BTC/USD`). When using a websocket connection you
+        always need to subscribe to the events you want to receive. You can do this
+        by using the `symbols` argument shown above or if your connection is already
+        created you can also use the `subscribe()` function to get data for additional
+        symbols. Example of adding symbols after the connection is already created:
+        ```python
+        ws = td.websocket(symbols=["BTC/USD", "ETH/USD"], on_event=on_event)
+        ws.connect()
+        ws.keep_alive()
+        ws.subscribe(["USDT/USD", "USDC/USD"])
+        ```
+    
+    You can also notice there's a line in the code example to keep the connection
+    alive *forever:*  `ws.keep_alive()`. It makes sure the connection will stay
+    active until it gets terminated. If you don't add this line the connection
+    might break instantly.
```

Comment:
Can we move this section before or after the procedure? There's a lot of concepts in here that don't really belong.

Reply:
https://github.com/timescale/docs/pull/1101/commits/847541b5633fe1936ff6a5f59473e1a162540fc6